### DISCUSSION TITLE
Close position in OMSType.HEDGING

### DISCUSSION THREAD

**Discussion by sidnvy** [nautechsystems/nautilus_trader#538] [Q&A] [answered]
Hi

In OMSType.HEDGING strategy, I'd like to place an order on specific side of futures position.
Take Binance for example ( I implemented binance futures api which similar to current nautilus_trader binance spot api).
### precondition:
Enabled `dualSidePosition` on binance.com (which match to Nautilus OMSType.HEDGING mode I think).
Currently hold 2 positions which are 0.01 BTC/USDT LONG and 0.01 BTC/USDT SHORT

### problem:
Now I want to close just SHORT position. Normally I will call `nautilus_trader.common.factories.OrderFactory`  limit, market, stop_market, etc... , but none of these factory methods provide the option for side of position, If I buy 0.01 BTC/USDT without specify the "SHORT" position side, it will increase my LONG position by default. 

So, how can NautilusTrader handle such situation? Should we consider add this kind of option for order factories?

**Comment by cjdsellers** [comment, marked as answer]
Hi @sidnvy 

You're very close to having that working by the sounds of it. You're correct that this `dualSidePosition` option is just another term for a hedging account type or `OMSType.HEDGING`.

We're currently working on docs and so some of this could be much clearer, however something to highlight is that the only responsibility of the `OrderFactory` is to instantiate orders in an easier way than if you had to pass every parameter into the order classes themselves.

The orders however have no knowledge of positions, even though by consequence they result in changes to positions when executed. 

**Solution**
So what you'd need to do is once you have your desired order from the factory, pass a `position_id` to the `submit_order` method in your strategy, along with the order. This is creating a command to submit that order specifically against that position (if it exists). e.g. the Cython definition:

```
cdef class SubmitOrder(TradingCommand):
    cdef readonly PositionId position_id
    """The position ID associated with the command.\n\n:returns: `PositionId` or ``None``"""
    cdef readonly Order order
    """The order for the command.\n\n:returns: `Order`"""
```

As you know implementing the Binance Futures integration is an open task for the project, and having a quick look at their API it seems that for the `/fapi/v1/order` endpoint there is a `positionSide` param. So in your execution client you'll need to do something like the following:
- When the `SubmitOrder` command arrives, you'll want to check if there is a `position_id` on that command. 
- If there is then get the position from the cache with `self.cache.position(position_id)`.
- If the position is `None` then log an error (cannot find position with ID).
- Then use the `position.side` value to determine what to pass as the `positionSide` param for the new order endpoint.

This should result in the order being submitted specifically against that position side for that market, and will either add or reduce to that position.

Hope that offers some clues for you. Feel free to continue the discussion and paste any code snippets if you like.

Let me know how it goes!

**Reply by sidnvy** [reply]
Really appreciate your solution here, now I understand the whole workflow even better!

Btw, I still not clear with `PositionSide.FLAT`, is that match with Binance One-way Mode or just simply means current position is flat (LONG+SHORT == 0)?
```cython
cpdef enum PositionSide:
    FLAT = 1
    LONG = 2
    SHORT = 3
```

**Reply by cjdsellers** [reply]
Correct, flat means neither long or short i.e. no position / out of the market.
